tpl/compare/truth.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 The Hugo Authors. All rights reserved.
+// The functions in this file is based on the Go source code, copyright
+// The Go Authors and  governed by a BSD-style license.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package compare provides template functions for comparing values.
+package compare
+
+import (
+	"reflect"
+
+	"github.com/gohugoio/hugo/common/hreflect"
+)
+
+// Boolean logic, based on:
+// https://github.com/golang/go/blob/178a2c42254166cffed1b25fb1d3c7a5727cada6/src/text/template/funcs.go#L302
+
+func truth(arg reflect.Value) bool {
+	return hreflect.IsTruthfulValue(arg)
+}
+
+// GetIf will return the given arg if it is considered truthful, else an empty string.
+// This is considered an internal template func, and probably not useful in
+// user-defined templates
+func (*Namespace) GetIf(arg reflect.Value) reflect.Value {
```

Comment:
This function is the only thing I don't like.  Can we copy this to `tpl/tplimpl` to avoid exporting this to the users?

Reply:
We can (and should, I guess) unexport it -- but we need it to be a template func, which will be available to the end users. But I think that if we make it private and don't document it anywhere + mark it as internal, we should be good.